app/bundles/LeadBundle/Helper/TokenHelper.php

```
@@ -110,6 +111,16 @@ public static function getValueFromTokens(array $tokens, $token)
      */
     private static function getTokenValue(array $lead, $alias, $defaultValue)
     {
+        // If a "vara-identifier" token is found try to update that patient with a unique identifier
+        if("vara-identifier" === $alias){
+            $vh = new VARAHelper();
+            if(isset($lead['varaid'])){
+                $patientVaraid = $lead['varaid'];
+                $patientIdentifier = $vh->savePatientUniqueIdentifier($patientVaraid);
+                return $patientIdentifier;
+            }
```

We should throw an error if there is no `varaid` contact field.